src/main/flight/pid.c

```diff
@@ -325,6 +339,16 @@ void pidController(const pidProfile_t *pidProfile, const rollAndPitchTrims_t *an
             currentPidSetpoint = pidLevel(axis, pidProfile, angleTrim, currentPidSetpoint);
         }
 
+        if (inCrashRecoveryMode && axis != FD_YAW) {
+            // self-level - errorAngle is deviation from horizontal
+            const float errorAngle =  -(attitude.raw[axis] - angleTrim->raw[axis]) / 10.0f;
+            currentPidSetpoint = errorAngle * levelGain;
+            if (cmpTimeUs(currentTimeUs, crashDetectedAtUs) > crashTimeLimitUs
+                || (ABS(attitude.raw[FD_ROLL] - angleTrim->raw[FD_ROLL]) < crashRecoveryAngleDeciDegrees
```

Comment:
crashGyroThreshold is probably huge - it must be high enough to trigger only when quad spins out of control.
Maybe rate corresponding to some percentage of stick deflection will work (but is unnecessarily complicated) - 10% deflection should be quite comfortable ...

Also it may be good idea to blend switchback a bit ..